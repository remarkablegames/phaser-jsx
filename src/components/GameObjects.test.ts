import Phaser from 'phaser';

import type { Props } from '../types';
import * as GameObjects from './GameObjects';

const keys = Object.keys(Phaser.GameObjects).filter(
  (key) =>
    typeof (Phaser.GameObjects as Record<string, unknown>)[key] ===
      'function' &&
    ![
      'GetCalcMatrix',
      'BuildGameObject',
      'BuildGameObjectAnimation',
      'GetTextSize',
      'MeasureText',
      'Sprite3D',
    ].includes(key),
);

it.each(keys)('exports %p', (key) => {
  expect((GameObjects as Props)[key]).toBe((Phaser.GameObjects as Props)[key]);
});

it('exports GameObjects', () => {
  expect(GameObjects).toMatchSnapshot();
});
