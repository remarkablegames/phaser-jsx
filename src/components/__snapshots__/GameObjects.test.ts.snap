// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`exports GameObjects 1`] = `
{
  "Arc": [Function],
  "BitmapText": [Function],
  "Blitter": [Function],
  "Bob": [Function],
  "Container": [Function],
  "Curve": [Function],
  "DOMElement": [Function],
  "DisplayList": [Function],
  "DynamicBitmapText": [Function],
  "Ellipse": [Function],
  "Extern": [Function],
  "GameObject": [Function],
  "GameObjectCreator": [Function],
  "GameObjectFactory": [Function],
  "Graphics": [Function],
  "Grid": [Function],
  "Group": [Function],
  "Image": [Function],
  "IsoBox": [Function],
  "IsoTriangle": [Function],
  "Layer": [Function],
  "Light": [Function],
  "LightsManager": [Function],
  "LightsPlugin": [Function],
  "Line": [Function],
  "Mesh": [Function],
  "NineSlice": [Function],
  "ParticleEmitter": [Function],
  "PathFollower": [Function],
  "Plane": [Function],
  "PointLight": [Function],
  "Polygon": [Function],
  "Rectangle": [Function],
  "RenderTexture": [Function],
  "Rope": [Function],
  "Shader": [Function],
  "Shape": [Function],
  "Sprite": [Function],
  "Star": [Function],
  "Text": [Function],
  "TextStyle": [Function],
  "TileSprite": [Function],
  "Triangle": [Function],
  "UpdateList": [Function],
  "Video": [Function],
  "Zone": [Function],
}
`;
